classes/user/UserInterest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * @file classes/user/UserInterest.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class UserInterest
+ *
+ * @brief UserInterest model calss
+ */
+
+namespace PKP\user;
+
+use APP\facades\Repo;
+use Eloquence\Behaviours\HasCamelCasing;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Query\JoinClause;
+use Illuminate\Database\Eloquent\Casts\Attribute;
+use Illuminate\Database\Eloquent\Builder;
+use Illuminate\Support\Collection;
+use PKP\core\ArrayItemIterator;
+use PKP\user\InterestEntryDAO;
+use PKP\db\DAORegistry;
+
+class UserInterest extends Model
+{
+    use HasCamelCasing;
+
+    public const CONTROLLED_VOCAB_INTEREST = 'interest';
+
+    protected $table = 'user_interests';
+    protected $primaryKey = 'user_interest_id';
+
+    protected $guarded = [
+        'user_interest_id',
+    ];
+
+    /**
+     * Indicates if the model should be timestamped.
+     *
+     * @var bool
+     */
+    public $timestamps = false;
+
+    protected function casts(): array
+    {
+        return [
+            'user_id' => 'integer',
+            'controlled_vocab_entry_id' => 'integer',
+        ];
+    }
+
+    /**
+     * Accessor and Mutator for primary key => id
+     */
+    protected function id(): Attribute
+    {
+        return Attribute::make(
+            get: fn($value, $attributes) => $attributes[$this->primaryKey] ?? null,
+            set: fn($value) => [$this->primaryKey => $value],
+        );
+    }
+
+    /**
+     * Compatibility function for including note IDs in grids.
+     *
+     * @deprecated 3.5.0 Use $model->id instead. Can be removed once the DataObject pattern is removed.
+     */
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    /**
+     * Compatibility function for including notes in grids.
+     *
+     * @deprecated 3.5. Use $model or $model->$field instead. Can be removed once the DataObject pattern is removed.
+     */
+    public function getData(?string $field): mixed
+    {
+        return $field ? $this->$field : $this;
+    }
+
+    /**
+     * Scope a query to only include notes with a specific assoc type and assoc ID.
+     */
+    public function scopeWithUserId(Builder $query, int $userId): Builder
+    {
+        return $query->where('user_id', $userId);
+    }
+
+    /**
+     * Get a list of controlled vocabulary entry IDs (corresponding to interest keywords) 
+     * attributed to a user
+     */
+    public static function getUserInterestIds(int $userId): array
```

Comment:
It is possible to centralise these (this one and below other static method) in the controlled vocab repo but these methods are only required for user interest . But they seems like out of place methods for a model . At the same time they are more like some generic method that should be accessible from a repo . Any thoughts ?

Reply:
I think this one is a bit of a unique case at the moment, since we don't have other relationship tables mapping `controlled_vocab_interest_id` to another entity. We might need them more in the future, but I don't mind if we don't refactor this to be reusable until that's needed.